File: app/src/main/java/rmnvich/apps/kinonika/domain/converter/BindingConverter.java
package rmnvich.apps.kinonika.domain.converter;

import android.databinding.BindingConversion;

import java.util.List;

import rmnvich.apps.kinonika.data.entity.Genre;
import rmnvich.apps.kinonika.data.entity.Tag;

public class BindingConverter {

    @BindingConversion
    public static String convertGenresToString(List<Genre> genres) {
        StringBuilder sb = new StringBuilder();
        for (Genre genre : genres) {
            if (sb.length() > 0) sb.append(", ");
            sb.append(genre.getName());
        }
        return sb.toString();
    }

    @BindingConversion
    public static String convertTagsToString(List<Tag> tags) {
        StringBuilder sb = new StringBuilder();
        for (Tag tag : tags) {
            if (sb.length() > 0) sb.append(" ");
            sb.append("#").append(tag.getHashTag());
        }
        return sb.toString();
    }
}

File: app/src/main/java/rmnvich/apps/kinonika/domain/converter/BindingConverter.kt
package rmnvich.apps.kinonika.domain.converter

import android.databinding.BindingConversion
import rmnvich.apps.kinonika.data.entity.Genre
import rmnvich.apps.kinonika.data.entity.Tag


class BindingConverter {

    @BindingConversion
    fun convertGenresToString(genres: List<Genre>): String {
        val sb = StringBuilder()
        for (genre in genres) {
            if (sb.isNotEmpty()) sb.append(", ")
            sb.append(genre.name)
        }
        return sb.toString()
    }

    @BindingConversion
    fun convertTagsToString(tags: List<Tag>): String {
        val sb = StringBuilder()
        for (tag in tags) {
            if (sb.isNotEmpty())
                sb.append(" ")
            sb.append("#" + tag.hashTag)
        }
        return sb.toString()
    }

}